### PR TITLE
[jit] Add len() for tuples

### PR DESCRIPTION
**Pull request by driazati**

### CODE REVIEW

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -1280,6 +1280,14 @@ RegisterOperators reg2({
       CREATE_LIST_OPS("Tensor", TensorList),
       CREATE_LIST_OPS("t", GenericList),
   #undef CREATE_LIST_OPS
+      Operator("aten::len(...) -> int", [](const Node* node) {
```
Comment (eellison) [inline torch/csrc/jit/register_prim_ops.cpp]: What is the ... syntax exactly ? Just for tuples ?

**Review by eellison — COMMENTED** [review]
Since the size of tuples is fixed, could you instead return the length by looking at the type input? 

& then add some sort of peephole optimization so that we don't evaluate the op at runtime

File: torch/csrc/jit/register_prim_ops.cpp (hunk 1 above)
Comment (driazati) [reply torch/csrc/jit/register_prim_ops.cpp]: It's for varargs so that this works for all tuples

File: torch/csrc/jit/register_prim_ops.cpp [hunk 2]
```diff
@@ -1280,6 +1280,17 @@ RegisterOperators reg2({
...
+        AT_ASSERT(node->inputs().size() == 1);
+        auto tuple = node->inputs().at(0)->type()->cast<TupleType>();
+        AT_ASSERT(tuple);
```
Comment (eellison) [inline torch/csrc/jit/register_prim_ops.cpp]: you can do type()->expect<TupleType>();

instead of ->cast && assert

**Pull request closed without merging**